### PR TITLE
[Bugfix]: Patch Auth Checks in Routing

### PR DESCRIPTION
**Pull request by Jibola**

### CODE REVIEW

File: {{cookiecutter.project_slug}}/frontend/app/lib/slices/authSlice.ts [hunk 1]
```diff
@@ -109,54 +113,58 @@ export const login =
...
-      dispatch(
-        addNotice({
-          title: "Login error",
-          content:
-            "Please check your details or internet connection and try again.",
-          icon: "error",
-        }),
-      )
-      dispatch(logout())
+    async (
```
Comment (NoahStapp) [inline {{cookiecutter.project_slug}}/frontend/app/lib/slices/authSlice.ts]: Are these just line number changes?

File: {{cookiecutter.project_slug}}/frontend/app/lib/slices/authSlice.ts [hunk 2]
```diff
@@ -109,54 +113,58 @@ export const login =
...
```
Comment (NoahStapp) [inline {{cookiecutter.project_slug}}/frontend/app/lib/slices/authSlice.ts]: Same here.

File: {{cookiecutter.project_slug}}/frontend/app/lib/slices/authSlice.ts [hunk 3]
```diff
@@ -208,99 +216,99 @@ export const createUserProfile =
...
-    const currentState = getState()
-    if (loggedIn(currentState) && currentState.tokens.access_token) {
-      try {
-        const res = await apiAuth.updateProfile(
-          currentState.tokens.access_token,
-          payload,
-        )
-        if (res) {
-          dispatch(setUserProfile(res))
+    async (dispatch: Dispatch, getState: () => RootState) => {
```
Comment (NoahStapp) [inline {{cookiecutter.project_slug}}/frontend/app/lib/slices/authSlice.ts:179]: Same here.

File: {{cookiecutter.project_slug}}/frontend/app/lib/slices/authSlice.ts [hunk 4]
```diff
@@ -208,99 +216,99 @@ export const createUserProfile =
...
-    const currentState = getState()
-    if (loggedIn(currentState) && currentState.tokens.access_token) {
-      try {
-        const res = await apiAuth.enableTOTPAuthentication(
-          currentState.tokens.access_token,
-          payload,
-        )
-        if (res) {
-          dispatch(setTOTPAuthentication(true))
+    async (dispatch: Dispatch, getState: () => RootState) => {
```
Comment (NoahStapp) [inline {{cookiecutter.project_slug}}/frontend/app/lib/slices/authSlice.ts:211]: Same here.

**Review by NoahStapp — CHANGES_REQUESTED** [review]
Looks good besides some line number changes that show up as a ton of new code, any idea what happened?

File: {{cookiecutter.project_slug}}/frontend/app/lib/slices/authSlice.ts (hunk 2 above)
Comment (Jibola) [reply {{cookiecutter.project_slug}}/frontend/app/lib/slices/authSlice.ts]: Yeah, these are just line-number changes. I'm actually going to curry them into the `handleGenericLogin` function, so they'll be gone as well.

File: {{cookiecutter.project_slug}}/frontend/app/lib/slices/authSlice.ts (hunk 3 above)
Comment (Jibola) [reply {{cookiecutter.project_slug}}/frontend/app/lib/slices/authSlice.ts:179]: same thing. looks like whitespace was changed. I'll file a task to run a linter on the repo.

File: {{cookiecutter.project_slug}}/frontend/app/lib/slices/authSlice.ts (hunk 4 above)
Comment (Jibola) [reply {{cookiecutter.project_slug}}/frontend/app/lib/slices/authSlice.ts:211]: see above

**Review by NoahStapp — APPROVED** [review]
LGTM!

**Pull request merged**